src/serializer/modules.js

```diff
@@ -98,21 +98,20 @@ class ModuleTracer extends Tracer {
         return undefined;
       }
 
-      // Even if we're not delaying unsupported requires, we want to record what
+      // If we're not delaying unsupported requires, we still want to record what
       // modules have been initialized
       if (!this.modules.delayUnsupportedRequires) {
         if (
           (this.requireStack.length === 0 || this.requireStack[this.requireStack.length - 1] !== moduleIdValue) &&
-          !this.modules.initializedModules.has(moduleIdValue) &&
           this.modules.moduleIds.has(moduleIdValue)
         ) {
           this.requireStack.push(moduleIdValue);
           try {
             let value = performCall();
             this.modules.initializedModules.set(moduleIdValue, value);
+            if (this.modules.initializedModules.has(moduleIdValue))
+              invariant(this.modules.initializedModules.get(moduleIdValue) === value);
```

Comment:
Let's add some more value to this invariant. Move the above `initializedModules.set` into an `else` branch of this `if`.